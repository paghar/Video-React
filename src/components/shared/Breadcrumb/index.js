//@flow
import React from 'react';
import { BreadcrumbWrapper, BreadcrumbItem, StyledIcon } from './index.style';
import Typography from 'components/shared/Typography';
import { Display1 } from 'theme/fonts';
import { Generals } from 'theme/colors';
type Props = {
  clicked: (id: number) => void,
  items: Array<string>,
  className?: string,
};
const Breadcrumb = ({ items, clicked, className }: Props) => {
  const listItems = items.map((name, id) => (
    <BreadcrumbItem key={id} onClick={() => clicked(id)}>
      {id > 0 && (
        <StyledIcon fontSize="11px" color={Generals.vi} iconNumber={61524} />
      )}
      <Typography font={Display1} color={Generals.vi}>
        {name}
      </Typography>
    </BreadcrumbItem>
  ));
  return (
    <BreadcrumbWrapper className={className}>{listItems}</BreadcrumbWrapper>
  );
};
export default Breadcrumb;
