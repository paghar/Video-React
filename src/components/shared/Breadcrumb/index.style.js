//@flow
import styled from 'styled-components';
import Icon from 'components/shared/Icon';
export const BreadcrumbWrapper = styled.ul`
  display: flex;
  display: box;
  list-style: none;
  align-items: flex-start;
  pointer-events: none;
  align-self: center;
  outline: none;
  border: none;
  background: none;
  padding-left: 0;
  padding-right: 0;
  margin: 0;
`;
export const BreadcrumbItem = styled.li`
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
`;
export const StyledIcon = styled(Icon)`
  padding: 0 13px;
`;
