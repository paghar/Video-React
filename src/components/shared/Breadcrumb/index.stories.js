//@flow
import React from 'react';
import { action } from '@storybook/addon-actions';
import Breadcrumb from '.';

export default { title: 'Breadcrumb' };
export const Breadcrumb3items = () => (
  <Breadcrumb
    clicked={action('clicked')}
    items={['Projects', 'Animal House', 'Episode1']}
  />
);
