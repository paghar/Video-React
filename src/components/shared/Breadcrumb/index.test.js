//@flow
import { shallow, mount } from 'enzyme';
import React from 'react';
import Breadcrumb from '.';
import renderer from 'react-test-renderer';

describe('<Breadcrumb />', () => {
  it('checks the UI', () => {
    const wrapper = renderer
      .create(
        <Breadcrumb
          clicked={console.log}
          items={['Projects', 'Animal House', 'Episode1']}
        />
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('checks click event', () => {
    const mockCallBack = jest.fn();
    const wrapper = mount(
      <Breadcrumb
        items={['Projects', 'Animal House', 'Episode1']}
        clicked={mockCallBack}
      />
    );
    wrapper
      .find('li')
      .last()
      .simulate('click');
    expect(mockCallBack).toHaveBeenCalled();
  });
  it('checks value of click event', () => {
    const mockCallBack = jest.fn();
    const wrapper = mount(
      <Breadcrumb
        items={['Projects', 'Animal House', 'Episode1']}
        clicked={mockCallBack}
      />
    );
    wrapper
      .find('li')
      .last()
      .simulate('click');
    expect(mockCallBack).toHaveBeenCalledWith(2);
  });
});
