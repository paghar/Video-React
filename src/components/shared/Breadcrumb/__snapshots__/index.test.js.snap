// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> checks the UI 1`] = `
.c3 {
  font-family: 'FontAwesome' !important;
  font-size: 11px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  display: box;
  list-style: none;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  pointer-events: none;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  outline: none;
  border: none;
  background: none;
  padding-left: 0;
  padding-right: 0;
  margin: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c4 {
  padding: 0 13px;
}

.c2 {
  pointer-events: auto;
}

<ul
  className="c0"
>
  <li
    className="c1"
    onClick={[Function]}
  >
    <span
      className="c2"
    >
      Projects
    </span>
  </li>
  <li
    className="c1"
    onClick={[Function]}
  >
    <span
      className="c3 c4"
      fontSize="11px"
    >
      
    </span>
    <span
      className="c2"
    >
      Animal House
    </span>
  </li>
  <li
    className="c1"
    onClick={[Function]}
  >
    <span
      className="c3 c4"
      fontSize="11px"
    >
      
    </span>
    <span
      className="c2"
    >
      Episode1
    </span>
  </li>
</ul>
`;
